presto-spi/src/main/java/com/facebook/presto/spi/schedule/NodeSelectionStrategy.java

```diff
@@ -17,5 +17,6 @@
 {
     HARD_AFFINITY,
     SOFT_AFFINITY,
+    SOFT_AFFINITY_PLUS_PLUS,
```

Comment:
I wonder if instead of adding a new `NodeSelectionStrategy` with a relatively opaque name (what does ++ mean?), this could be a node scheduler configuration flag that modified the behavior of the existing `SOFT_AFFINITY` strategy.

Reply:
I agree, I wonder if we could just leverage HARD_AFFINITY for it? As the current HARD_AFFINITY is more or less does not get used.

Reply:
I probably wouldn't go that route. Even if it's hardly used, it would be surprising to find that the `HARD_AFFINITY` policy would ever place a split on it's non-preferred node. I think this lives more comfortably as a `NodeSchedulerConfig` property that toggles "split cacheable on soft affinity fallback node" or something to that effect.